src/mailroom.py

```diff
@@ -0,0 +1,339 @@
+               'prompt_message':
+               'Options:' +
+               '\t[1] Create Report\n' +
+               '\t\t[2] Send a Thank\n' +
+               '\t\t[0] Quit the Program\n\n' +
+               'Enter Command: ',
+               'valid_responses': state_1_valid_responses,
+               }
+    state_2 = {'state': 2,
+               'comment': 'This is the create a thank you letter state.',
+               'prompt_message':
+               'Options: \n' +
+               '\t\t[l or list] Type "l" or list to get a list of current donors\n' +
+               '\t\t[name]      Enter a donor name to write a thank you letter\n' +
+               '\t\t[0]         Return to Main Menu\n\n' +
+               'Enter Command or donor name: ',
+               'user_response': 'default_response',
+               'valid_responses': state_2_valid_responses,
+               }
+    state_3 = {'state': 3,
+               'comment': 'This is Print a Report of donor information.',
+               'prompt_message': '',
+               'user_response': '',
+               'valid_responses': state_3_valid_responses,
+               }
+    state_4 = {'state': 4,
+               'comment': 'Enter New Donation Amount',
+               'prompt_message': 'Enter New Donation Amount or' +
+               ' 0 to return to previous menu: ',
+               'donation_amount': 0.00,
+               'valid_responses': state_4_valid_responses,
+               }
+    state_5 = {'state': 5,
+               'comment': 'Ask to print a Thank You Letter',
+               'prompt_message': 'Print a thank you letter? (y or n): ',
+               'valid_responses': state_5_valid_responses,
+               }
+    state_6 = {'state': 6,
+               'comment': 'Ask to validate creating a user',
+               'prompt_message': ' (y or n): ',
+               'valid_responses': state_6_valid_responses,
+               'new_donor_name': ''
+               }
+    state_7 = {'state': 7,
+               'comment': 'Ask to validate donation amount',
+               'prompt_message': ' (y or n): ',
+               'valid_responses': state_7_valid_responses,
+               'new_donor_name': ''
+               }
+
+    state_dict = {
+        'state_1': state_1,
+        'state_2': state_2,
+        'state_3': state_3,
+        'state_4': state_4,
+        'state_5': state_5,
+        'state_6': state_6,
+        'state_7': state_7,
+    }
+
+    return state_dict
+
+
+def state_1_valid_responses(a):
+    global CURRENT_STATE
+    global STATE_DICT
+    if a == '0':
+        os.system('clear')
+        print('Good By')
+        sys.exit()
+    elif a == '1':
+        create_report()
+        CURRENT_STATE = STATE_DICT['state_1']
+        return True
+    elif a == '2':
+        CURRENT_STATE = STATE_DICT['state_2']
+        os.system('clear')
+        return True
+    else:
+        print('That was not a valid input')
+        return False
```

Comment:
So, in this case, you are returning True/False to signal that things were okay.  At the same time, you are altering a global "CURRENT_STATE" in order to control what happens next.  Why do both?

If you return the next state you want to enter, then returning None, or better, the current state itself, is enough of a signal that things didn't validate.  After all, the same run-state will be entered, which means that the same user prompt can be used.

If things are going well, then returning the new state allows you to rebind the state in the running context without needing to update a global state variabe.